packages/server/src/restaurants/profile/domain/UpdatePartnerProfileHandler.ts

```diff
@@ -20,19 +21,27 @@ class UpdatePartnerProfileHandler implements Handler<UpdatePartnerProfileRequest
   constructor(@InjectModel(Restaurant.name) private restaurantModel: Model<RestaurantDocument>) {}
 
   async exec(req: UpdatePartnerProfileRequest): Promise<null | undefined> {
-    const result = await this.restaurantModel.findOneAndUpdate(
-      { _id: req.id },
-      {
-        name: req.name,
-        description: req.description,
-        cuisineType: req.cuisineType,
-        tags: req.tags,
-        bankAccountNumber: req.bankAccountNumber,
-        phoneNumber: req.phoneNumber,
-      },
-    );
+    const filter = { _id: req.id };
+    const update = {
+      name: req.name || null,
+      description: req.description || null,
+      cuisineType: req.cuisineType || null,
+      tags: req.tags || null,
+      bankAccountNumber: req.bankAccountNumber || null,
+      phoneNumber: req.phoneNumber || null,
+    };
+
+    const result = await this.restaurantModel.findOneAndUpdate(filter, update, { new: true });
 
     if (result === null) return null;
+
+    const isNotNull = (val) => val != null;
+
+    if (!_.values(result?.toObject()).some(_.isNull)) {
+      result.profileCompleted = true;
+      result.save();
```

Comment:
Tutaj brakuje `await`, chociaż z drugiej strony, przeniósłbym całe sprawdzanie czy `profileComplete` do modelu Partner np. jako metoda `isCompleted()`, wtedy nie trzeba by trzymać w bazie boolean `profileCompleted`.

Reply:
@htk4 w jakim pliku miałaby się znajdować po przeniesieniu ta metoda ?

Reply:
Btw też mamy to pole dla `user` to rozumiem, że w takim wypadku żeby było wszędzie tak samo to tam też trzeba będzie zmienić.

Reply:
> Btw też mamy to pole dla `user` to rozumiem, że w takim wypadku żeby było wszędzie tak samo to tam też trzeba będzie zmienić.

No fajnie byłoby tak zrobić, ale to może do osobnego taska.

Reply:
> @htk4 w jakim pliku miałaby się znajdować po przeniesieniu ta metoda ?

Myślę, że najłatwiej byłoby ją przenieść do [RestaurantSchema](https://github.com/CodersCamp2021-HK/CodersCamp2021.Project.Fullstack/blob/main/packages/server/src/restaurants/database/RestaurantSchema.ts) jako metodę [virtualną](https://mongoosejs.com/docs/tutorials/virtuals.html). Wtedy w `RestaurantSchema` stworzyło by się zwykły getter (bez dekoratorów):
```ts
class Restaurant {
  get isCompleted() {
    // Tutaj logika sprawdzania czy profile completed
    return true;
  }
}
```
Później żeby ten getter był widoczny z poziomu documentu mongoose trzeba go podpiąć do schemy:
```ts
RestaurantSchema.virtual('isCompleted').get(Restaurant.prototype.isCompleted);
```